File: web/src/types/external/themoviedb/Person.ts
export interface Person {
  adult?: boolean;
  also_known_as?: object[];
  biography?: string;
  birthday?: string;
  deathday?: string;
  gender?: number;
  homepage?: string;
  id: number;
  imdb_id?: string;
  name?: string;
  place_of_birth?: string;
  popularity?: number;
  profile_path?: string;
  combined_credits?: PersonCredits;
}

export interface PersonCredits {
  cast?: PersonCredit[];
  crew?: PersonCredit[];
}

export interface PersonCredit {
  adult?: boolean;
  backdrop_path?: string;
  genre_ids?: number[];
  id: number;
  character?: string;
  media_type?: string;
  original_language?: string;
  original_title?: string;
  overview?: string;
  popularity?: number;
  poster_path?: string;
  release_date?: string;
  name?: string;
  title?: string;
  video?: boolean;
  vote_average?: number;
  vote_count?: number;
}

export interface CastMember {
  character?: string;
  credit_id?: string;
  gender?: number;
  id: number;
  name?: string;
  order?: number;
  profile_path?: string;
}

export interface CrewMember {
  credit_id?: string;
  department?: string;
  gender?: number;
  id: number;
  job?: string;
  name?: string;
  profile_path?: string;
}

File: backend/common/src/main/scala/com/teletracker/common/model/tmdb/Person.scala
package com.teletracker.common.model.tmdb

import com.teletracker.common.util.Slug
import io.circe.generic.JsonCodec
import io.circe._
import io.circe.shapes._
import io.circe.generic.semiauto._
import io.circe.syntax._
import com.teletracker.common.util.json.circe._
import shapeless.{:+:, CNil}

trait PersonId

@JsonCodec
case class Person(
  adult: Option[Boolean],
//  also_known_as: Option[Map[String, Any]],
  biography: Option[String],
  birthday: Option[String],
  deathday: Option[String],
  gender: Option[Int],
  homepage: Option[String],
  id: Int,
  imdb_id: Option[String],
  name: Option[String],
  place_of_birth: Option[String],
  popularity: Option[Double],
  profile_path: Option[String],
  known_for: Option[List[Movie :+: TvShow :+: CNil]],
  // Join fields
  credits: Option[PersonCredits],
  combined_credits: Option[PersonCredits],
  movie_credits: Option[PersonMovieCredits],
  tv_credits: Option[PersonTvCredits])
    extends TmdbQueryableEntity
    with HasTmdbId

@JsonCodec case class CastMember(
  character: Option[String],
  character_name: Option[String],
  credit_id: Option[String],
  gender: Option[Int],
  id: Int,
  name: Option[String],
  order: Option[Int],
  profile_path: Option[String],
  // Computed fields
  normalizedName: Option[Slug] = None)

@JsonCodec case class CrewMember(
  credit_id: Option[String],
  department: Option[String],
  gender: Option[Int],
  id: Int,
  job: Option[String],
  name: Option[String],
  profile_path: Option[String])

@JsonCodec case class PersonMovieCredits(
  crew: List[Movie],
  cast: List[Movie])

@JsonCodec case class PersonTvCredits(
  crew: List[TvShow],
  cast: List[TvShow])

@JsonCodec
case class PersonCredits(
  crew: List[PersonCredit],
  cast: List[PersonCredit])

@JsonCodec
case class PersonCredit(
  adult: Option[Boolean],
  backdrop_path: Option[String],
  genre_ids: Option[List[Int]],
  department: Option[String],
  id: Int,
  character: Option[String],
  media_type: Option[MediaType],
  job: Option[String],
  original_language: Option[String],
  original_title: Option[String],
  overview: Option[String],
  popularity: Option[Double],
  poster_path: Option[String],
  release_date: Option[String],
  name: Option[String],
  title: Option[String],
  video: Option[Boolean],
  vote_average: Option[Double],
  vote_count: Option[Int])
